src/index.js

```diff
@@ -17,16 +18,18 @@ export function useMemoOne<T>(
   const initial: Cache<T> = useState(() => ({
     inputs,
     result: getResult(),
+    getResult,
   }))[0];
 
   const committed = useRef<Cache<T>>(initial);
 
   // persist any uncommitted changes after they have been committed
 
   const isInputMatch: boolean = Boolean(
-    inputs &&
-      committed.current.inputs &&
-      areInputsEqual(inputs, committed.current.inputs),
+    inputs
+      ? committed.current.inputs &&
+          areInputsEqual(inputs, committed.current.inputs)
+      : getResult === initial.getResult,
```

Comment:
When `inputs` is undefined, this compares `getResult` and `initial.getResult`, which are always equal on the first render. After the first render, they are always *not* equal, unless you wrapped the `getResult` argument with `useCallback` or `useMemo` (which nobody does, right?) or you hoisted the `getResult` function out of the component (so its identity never changes between renders).

If those two edge cases are too sketchy, we could use a nonce instead. On each `useMemoOne` call, the nonce would be incremented, to ensure `initial.nonce` is equal to the current `nonce` on the first render only.